svsim/src/main/scala/vcs/Backend.scala

```diff
@@ -96,11 +96,60 @@ object Backend {
     assertionSettings:      Option[AssertionSettings] = None
   )
 
+  /** Settings for controlling VCS coverage.
+    *
+    * These options map to the `-cm` option.  Each parameter turns on a specific
+    * kind of coverage.
+    *
+    * @param line enable line coverage
+    * @param cond enable condition coverage
+    * @param fsm enable finite-state-machine coverage
+    * @param tgl enable toggle coverage
+    * @param branch enable branch coverage
+    * @param assert enable assert coverage
+    */
+  // Note: This case class is being clever and re-using the names of the
+  // elements as the names of the options.  This makes these very terse, but
+  // they then match, exactly, the documentation in the VCS manual.
+  final case class CoverageSettings(
+    line:   Boolean = false,
+    cond:   Boolean = false,
+    fsm:    Boolean = false,
+    tgl:    Boolean = false,
+    branch: Boolean = false,
+    assert: Boolean = false
+  ) {
+    def compileFlags: Seq[String] = productElementNames.zip(productIterator).flatMap {
+      case (_, false)   => None
+      case (name, true) => Some(name)
+    } match {
+      case coverageTypes if !coverageTypes.isEmpty =>
+        // Note: It is unclear how many of these other options are actually
+        // necessary.  These options were migrated from internal flows which
+        // were cargo culted from other internal flows.
+        //
+        // TODO: Clean this up such that each option is broken out into an
+        // actual VCS option.  E.g., `-cm_tgl` should be its own option.  E.g.,
+        // if you want all these options this should be handled with an omnibus
+        // setting.
+        (Seq(
+          "-cm"
+        ) :+ coverageTypes.mkString("+") :+
+          "+define+FIRRTL_BEFORE_INITIAL=// VCS coverage off" :+
+          "+define+FIRRTL_AFTER_INITIAL=// VCS coverage on" :+
+          "+define+USE_UNR_ONLY_CONSTRAINTS") ++ Option.when(line || tgl)("-cm_seqnoconst") ++
+          Option.when(tgl)(Seq("-cm_tgl", "portsonly")).toSeq.flatten ++
+          Option.when(branch)(Seq("-cm_branch", "ignoreMissingDefault")).toSeq.flatten
```

Comment:
This confuses me a bit, so if `tgl` is `true`, we you get both: `-cm +tgl` and `-cm_cgl portsonly`?

Reply:
It confused me, too... However, this is the set of internal options.

How about this? I'll break these out into individual options and then bridge them internally.

Reply:
I backed this out and the options now are transparently mapped to their VCS options in https://github.com/chipsalliance/chisel/pull/4793/commits/9559742ab5bd4042c3a75b36ba692458a795a3cc.

A consideration is that we generally do not expose options this way for Chisel/firtool. Those, we happily make them `Array[String]` and let the user get it correct. With svsim, we already started down the path of type-safe options. Given the large amount of VCS options, we can either add these on demand as needed, or we could switch to the `Array[String]` approach for backend options.

There is some benefit to the `Array[String]` approach in that backends no longer have different option types. However, it's more stringy.

Reply:
I honestly would prefer less string-y arguments for Chisel and firtool, but am certainly not going to let the perfect be the enemy of the good on doing that for VCS coverage options and just leaving the Chisel and firtool options as `Array[String]` for the time being.

One tricky bit about structured types is that you probably want some form of parsing into your type which using `Array[String]` allows you to gloss over (this was a nice thing about how we/you did options annotations), but again, I still think this is good as is.